src/equations/shallow_water_two_layer_2d.jl

```diff
@@ -0,0 +1,936 @@
+
+  # Calculate averaged velocity of both layers
+  v1_m = (h_v1_upper + h_v1_lower) / (h_upper + h_lower)
+  v2_m = (h_v2_upper + h_v2_lower) / (h_upper + h_lower)
+
+  h_upper, h_lower = waterheight(u, equations)
+  v1_upper, v2_upper, v1_lower, v2_lower = velocity(u, equations)
+
+  c = sqrt(equations.gravity * (h_upper + h_lower)) 
+  return (max(abs(v1_m) + c, abs(v1_upper), abs(v1_lower)), 
+          max(abs(v2_m) + c, abs(v2_upper), abs(v2_lower)))
+end
+
+
+# Helper function to extract the velocity vector from the conservative variables
+@inline function velocity(u, equations::ShallowWaterTwoLayerEquations2D)
+  h_upper, h_v1_upper, h_v2_upper, h_lower, h_v1_lower, h_v2_lower, _ = u
+
+  v1_upper = h_v1_upper / h_upper
+  v2_upper = h_v2_upper / h_upper
+  v1_lower = h_v1_lower / h_lower
+  v2_lower = h_v2_lower / h_lower
+
+  return v1_upper, v2_upper, v1_lower, v2_lower
+end
+
+
+# Convert conservative variables to primitive
+@inline function cons2prim(u, equations::ShallowWaterTwoLayerEquations2D)
+  h_upper, _, _, h_lower, _, _, b = u
+
+  H_lower = h_lower + b
+  H_upper = h_lower + h_upper + b
+  v1_upper, v2_upper, v1_lower, v2_lower = velocity(u, equations)
+
+  return SVector(H_upper, v1_upper, v2_upper , H_lower, v1_lower, v2_lower, b)
+end
+
+
+# Convert conservative variables to entropy variables
+# Note, only the first four are the entropy variables, the fifth entry still just carries the bottom
+# topography values for convenience. 
+# In contrast to general usage the entropy variables are denoted with q instead of w, because w is
+# already used for velocity in y-Direction
+@inline function cons2entropy(u, equations::ShallowWaterTwoLayerEquations2D)
+  h_upper, _, _, h_lower, _, _, b = u
+  # Assign new variables for better readability
+  rho_upper = equations.rho_upper
+  rho_lower = equations.rho_lower
+  v1_upper, v2_upper, v1_lower, v2_lower = velocity(u, equations)
+
+  w1 = rho_upper * (equations.gravity * (              h_upper + h_lower + b) +
+               - 0.5 * (v1_upper^2 + v2_upper^2))
+  w2 = rho_upper * v1_upper
+  w3 = rho_upper * v2_upper
+  w4 = rho_lower * (equations.gravity * (equations.r * h_upper + h_lower + b) +
+               - 0.5 * (v1_lower^2 + v2_lower^2))
+  w5 = rho_lower * v1_lower
+  w6 = rho_lower * v2_lower
+  return SVector(w1, w2, w3, w4, w5, w6, b)
+end
+
+
+# Convert primitive to conservative variables
+@inline function prim2cons(prim, equations::ShallowWaterTwoLayerEquations2D)
+  H_upper, v1_upper, v2_upper, H_lower, v1_lower, v2_lower, b = prim
+
+  h_lower = H_lower - b
+  h_upper = H_upper - h_lower - b
+  h_v1_upper = h_upper * v1_upper
+  h_v2_upper = h_upper * v2_upper
+  h_v1_lower = h_lower * v1_lower
+  h_v2_lower = h_lower * v2_lower
+  return SVector(h_upper, h_v1_upper, h_v2_upper, h_lower, h_v1_lower, h_v2_lower, b)
+end
+
+
+@inline function waterheight(u, equations::ShallowWaterTwoLayerEquations2D)
+  return u[1], u[4]
```

Comment:
```suggestion
  return SVector(u[1], u[4])
```

Reply:
Done in 4f2114ca248344248107f39fd2d7e0523083b079